Update ff version to v88.0

trickypr opened the pull request:
This pull request updates ff to v88, fixing dothq/browser-desktop#108 and dothq/browser-desktop#106. **This pull request will bork the code base if you merge it now**. Before merging the following must be changed
- [x] Builds should work
- [x] Some basic ui bugs (e.g. close button sizes) should be fixed
- [x] Reviewers should be comfortable with there not being any critical bugs.

Note that some changes have been made to the code base. Those are listed in dothq/browser-desktop#111. I am going to be off the internet for another 2 weeks (apart from Fridays) because exams are coming up, so feel free to make changes to this pull request to get it working.

kierandrewett committed:
🍉 Add --hard option to reset

kierandrewett committed:
🌐 Add hardWarning logging option

dothq-robot committed:
🎨 Enforce code style

kierandrewett committed:
✨ Add hard reset logic

kierandrewett committed:
💬 Let user know we're deleting the src dir

kierandrewett committed:
Merge branch '88.0' of https://github.com/dothq/browser-ff into 88.0

dothq-robot committed:
🎨 Enforce code style

kierandrewett committed:
🆙 Bump Firefox version to 88.0

kierandrewett committed:
💄 Add clean progress bar for melon download

kierandrewett committed:
🐛 Kill host after finishing download

kierandrewett committed:
Merge branch '88.0' of https://github.com/dothq/browser-ff into 88.0

dothq-robot committed:
🔧 Update current build changeset

kierandrewett committed:
Merge pull request #29 from dothq/nightly

add missing changes

dothq-robot committed:
🎨 Enforce code style

kierandrewett committed:
:see_no_evil: Hide .dotbuild directory from VSC listing

kierandrewett committed:
Merge branch '88.0' of https://github.com/dothq/browser-desktop into 88.0

kierandrewett committed:
Merge pull request #30 from dothq/nightly

pull in changes

dothq-robot committed:
🎨 Enforce code style

kierandrewett committed:
🆙 Update firefox.js patch to Firefox 88.0

kierandrewett committed:
🆙 Update browser-context.inc patch for Firefox 88.0

kierandrewett committed:
:up: Update browser.xhtml patch to Firefox 88.0

kierandrewett committed:
:up: Update BrowserGlue.jsm patch to Firefox 88.0

kierandrewett committed:
:up: Update CustomizableUI.jsm patch to Firefox 88.0

kierandrewett committed:
:up: Update aboutwelcome.jsx patch to Firefox 88.0

kierandrewett committed:
:up: Update searchbar.js patch to Firefox 88.0

kierandrewett committed:
:up: Update moz.build patch to Firefox 88.0

kierandrewett committed:
:up: Update moz.build patch to Firefox 88.0

kierandrewett committed:
:up: Update screenshot.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update browser.css patch to Firefox 88.0

kierandrewett committed:
:up: Update browser.css patch to Firefox 88.0

kierandrewett committed:
:up: Update folder.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update indicator-private-browsing.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update security.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update jar.inc.mn patch to Firefox 88.0

kierandrewett committed:
:up: Update menupanel.inc.css patch to Firefox 88.0

kierandrewett committed:
:up: Update reader-mode.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update browser.css patch to Firefox 88.0

kierandrewett committed:
:up: Update moz-input-box.js patch to Firefox 88.0

kierandrewett committed:
:up: Update LightweightThemeConsumer.jsm patch to Firefox 88.0

kierandrewett committed:
:up: Update close.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update common.inc.css patch to Firefox 88.0

kierandrewett committed:
😁 Add update patch gitmoji

kierandrewett committed:
💄 Fix loading wrong theme

kierandrewett committed:
🐛 Fix syntax error in browser.xhtml

kierandrewett committed:
🔥 Remove duplicate constant in customizableui

kierandrewett committed:
🆙 Update patches

kierandrewett committed:
💄 Fix padding of menuitems

kierandrewett committed:
Merge branch 'nightly' into 88.0

dothq-robot committed:
🎨 Enforce code style

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🚨 Fix patches
These still have major issues, see: dothq/browser#583

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🐛 Fix launch error

trickypr committed:
Merge branch '88.0' of https://github.com/dothq/browser-desktop into 88.0

trickypr committed:
🩹 Revert changes to top sites
See: dothq/browser#583

trickypr committed:
🐛 Remove alpenglow
See dothq/browser#583

trickypr committed:
💄 Reapply page security standard
See dothq/browser#583

trickypr committed:
Merge branch '88.0' into v88-staging

dothq-robot committed:
🎨 Enforce code style

trickypr committed:
🔀 Merge nightly changes into v88 (#35)

🔀 Merge nightly changes into v88

trickypr committed:
🆙 Fix patches

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🆙 Reexport patches

trickypr committed:
💄 Correct close button size

dothq-robot committed:
🔧 Update current build changeset

trickypr commented:
I believe this pull request matches the quality of nightly and is ready for review.

trickypr committed:
💄 Restore folder and defaultFavicon icons

trickypr commented:
The identity-icon file doesn't appear to be present in the ff code-base under the same name: https://searchfox.org/mozilla-central/search?q=identity-icon

**Edit:** It now uses `info`

trickypr committed:
🔀 Merge branch '88.0' of https://github.com/dothq/browser-desktop into 88.0

trickypr committed:
💄 Add some more icons

trickypr commented:
@EnderDev I need a `security-warning` icon to finish this pull request, then it should be ready to open up for review

trickypr committed:
💄 Add Dot Icons for security-warning and security-broken
dothq/browser#583

dothq-robot committed:
🔧 Update current build changeset

trickypr commented:
All of the issues in dothq/browser-desktop#111 have been addressed, I am closing that issue.

trickypr commented:
Dot browser is now 2 versions behind ff. Firefox v89 has been released

trickypr commented on l10n/en-US/browser/aboutCertError.ftl:
```diff
@@ -1,5 +0,0 @@
-# This Source Code Form is subject to the terms of the Mozilla Public
```
Is it fine that the localization files were deleted by melon?

kierandrewett replied:
Yes, I have fixed them and converted them back to patch files.

trickypr commented on patches/browser-base-content-nsContextMenu-js.patch:
```diff
@@ -49,38 +49,24 @@ index b03abc5d09fb1a91795d2b6c5433a6f548881f52..f77cefe7954dd2238e616a7e6f343341
+         nsContextMenu.DevToolsShim.isDevToolsUser());
+ 
+     this.showItem("context-viewsource", shouldShow);
++//     Something changed here with ff88. This line was removed. Idk why, so it has been commented out
```
This changed in ff88 and I am not sure what to do about it.

kierandrewett replied:
Find where context-viewinfo is and make sure you add to the disabled attr `gIdentityHandler._hasInvalidPageProxyState()`

trickypr replied:
The issue is that `context-viewinfo` appears to have been removed or renamed: https://searchfox.org/mozilla-central/search?q=context-viewinfo&path=

trickypr replied:
Ok, the `View page info` option is gone in ff89, so I am going to assume they removed it in v88.
<img width="236" alt="Screen Shot 2021-06-04 at 3 20 10 pm" src="https://user-images.githubusercontent.com/23250792/120749690-8ce7d500-c4f4-11eb-8b25-d6987b44da03.png">

trickypr replied:
I can confirm that this was removed in v88

trickypr committed:
🎨 Correctly format package.json

trickypr committed:
➕ Update execa

trickypr committed:
🔥 Remove context-viewinfo code
The "View page info" option has been removed from firefox

trickypr committed:
Merge branch '88.0' into v88-staging

trickypr committed:
🔀 Merge pull request #37 from dothq/v88-staging

🔀 Merge nightly into v88

trickypr committed:
🆙 Fix patches

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🆙 Regenerate patch files

trickypr committed:
Merge branch '88.0' of https://github.com/dothq/browser-desktop into 88.0

trickypr committed:
🆙 Exclude onboarding icons
Onbording icons folder do not seem to be present yet. Once you add them uncomment the lines commented here.

dothq-robot committed:
🔧 Update current build changeset

drmonocle-official commented:
Hello! 

I just wanna ask how do you made the browser based on Mozilla browser... 
I would like to know and maybe I could help you with Dot!

Please mention me when you answer.
Thank you!

trickypr commented:
This pull request is being closed in favor of migrating straight to v89. The branch will remain for reference until we merge v89 into nightly
